### ISSUE TITLE
train, test and validation sets with im2rec

### ISSUE THREAD

**Issue by patriciaaa82** [apache/incubator-mxnet#6614]
Hi,
I want to split my data into train, validation and test sets. 
I'm using im2rec.py to generate the .rec files with the following commands:

python im2rec.py --list=1 --recursive=1 --shuffle=1 --test-ratio=0.3 --train-ratio=0.7 images/images images/
python im2rec.py --num-thread=4 --pass-through=1 images/images images

Which gives me a train.rec and test.rec. But how can I get also a validation set?
Thanks

**Comment by ysh329** [comment]
You can put the validation set in another directory. My means is using `im2rec.py` and setting --train-ratio=1.0 as validation set.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!

**Comment by anujgupta82** [comment]
@patriciaaa82 : I am looking for the exact same thing
Did you get the answer?

**Comment by chitraa** [comment]
@patriciaaa82 It is supported. You can use the command
`python im2rec.py --list=1 --recursive=1 --shuffle=1 --test-ratio=0.1 --train-ratio=0.8 images/images images/` for 80% train, 10% test and the remaining 100-80-10 = 10% validation.
Here is a snippet from im2rec.py which does the same:
`            if args.test_ratio:
                write_list(args.prefix + str_chunk + '_test.lst', chunk[:sep_test])
            if args.train_ratio + args.test_ratio < 1.0:
                write_list(args.prefix + str_chunk + '_val.lst', chunk[sep_test + sep:])
            write_list(args.prefix + str_chunk + '_train.lst', chunk[sep_test:sep_test + sep])`
